Burak stripe integration

myniqx opened the pull request:
it isnt over yet.

7alip commented on apps/hvn/package.json:
```diff
@@ -21,19 +22,28 @@
     "react-icons": "^5.5.0",
     "sharp": "0.33.5",
+    "sonner": "^2.0.1",
     "stripe": "^17.6.0",
```
https://payloadcms.com/docs/plugins/stripe

@myniqx I thought that this plugin helps with integrating.

7alip commented on apps/hvn/src/components/donation/stripe-form.tsx:
```diff
@@ -0,0 +1,169 @@
+
+import { Button } from '@fc/ui/base/button'
+import { Card, CardContent, CardHeader, CardTitle } from '@fc/ui/base/card'
+import { Input } from '@fc/ui/base/input'
+import { Label } from '@fc/ui/base/label'
```
@myniqx I removed local base components and import them from @fc/ui. Did you install these base components because the hvn app may be moved to a separate project?

myniqx replied:
we have already seperated from base project. dont you remember ?

7alip commented on apps/hvn/src/app/api/webhook/route.ts:
```diff
@@ -0,0 +1,44 @@
+    const tempBody = JSON.parse(body)
+    console.log({ body: { ...tempBody, data: JSON.stringify(tempBody.data, null, 2) } })
+
+    if (event.type === 'payment_intent.succeeded') {
```
Do we need to add other event types to show relevant information in the UI?

<img width="1845" alt="Screenshot 2025-02-23 at 11 25 01" src="https://github.com/user-attachments/assets/58ddb32b-0813-417a-83b3-c9a9f3ab22c7" />

MustafaTRKYLMZ replied:
As a type, we have only one event. This website is special for hvn. It means we have only one type of event in this website. But We can have show different locations (Amsterdam, Utrecht, Den Haag)

7alip replied:
@myniqx It seems plugin doesn't support the latest Stripe version.

<details><summary>Details</summary>

<img width="1373" alt="Screenshot 2025-02-24 at 09 01 08" src="https://github.com/user-attachments/assets/07d65fbc-fd9f-47a8-b9c4-0510d7fb448e" />
</details>

myniqx replied:
> Do we need to add other event types to show relevant information in the UI?

mostly we dont. there are 4 packets comes after a payment. most of them are redundant for us. 

> As a type, we have only one event. This website is special for hvn. It means we have only one type of event in this website. But We can have show different locations (Amsterdam, Utrecht, Den Haag)

this is stripe payment result event :) failed succeded etc

>  It seems plugin doesn't support the latest Stripe version.

i couldnt make payloads plugins webhook work. it is triggered by strapi. but somehow plugin wont run my webhook function at all. maybe because of version. but i keep those files, in case payloads plugin updated.

7alip commented on apps/hvn/src/components/donation/donation-button.tsx:
```diff
@@ -0,0 +1,50 @@
+  }, [status])
+
+  const handleDonate = async () => {
+    const successUrl = `${window.location.origin}${path}?status=success`
+    const cancelUrl = `${window.location.origin}${path}?status=cancel`
```
Would it be better to create a separate page for payment status? In this case we can move the status logic from homepage.

MustafaTRKYLMZ commented on apps/hvn/src/components/support/support-bank.tsx:
```diff
@@ -0,0 +1,49 @@
+        </div>
+
+        <p className="text-sm text-muted-foreground">
+          Sayfamızı size daha iyi hizmet sunmak için sürekli olarak geliştiriyoruz.
```
TODO: Translate should be add

myniqx commented:
@7alip  i didnt know that payload has stripe plugin. as i pushed this branch i was hoping i can have those kind informations. but making commit while i still working, now i have confilicts to deal with :(

MustafaTRKYLMZ commented on apps/hvn/src/components/support/support-bank.tsx:
```diff
@@ -0,0 +1,49 @@
+          <img alt="Stichting Logo" className="h-16 w-16 rounded-full object-cover shadow-md" src="/images/logo.png" />
+          <div>
+            <h3 className="text-xl font-bold">Harmonie van Nederland</h3>
+            <p className="text-sm text-muted-foreground">(Stichting Wees de Stem voor Vrijheid)</p>
```
It is not necessary. Because this project is common. Freedom Combination and SANAT (There are two foundation in there)

But for donation info, we can add this sentence: 

This is a joint project of the SANAT and the Stg. Wees de stem voor vrijheid Foundation (Freedom Combination), but donations will be managed by the Freedom Combination.

myniqx commented:
@7alip  i added plugin. but there is a problem. after i install plugin, when i request donation, on chrome's network says fetch canceled then page refreshes nothing happens. i thought it is plugin. but i removed it and installed stripe back. it still happens. when i remove node_modules and reinstall it then one time it works. last night while i write webhook function, that button works without any problem repeadetly. can you check it?

7alip commented:
@myniqx I reproduced it today as well. It's not because of the plugin but maybe the button itself. We can try changing type of the button, submit/button. Otherwise I'll check tomorrow

7alip commented on apps/hvn/package.json:
```diff
@@ -20,21 +21,31 @@
     "react-hook-form": "^7.54.2",
     "react-icons": "^5.5.0",
     "sharp": "0.33.5",
-    "stripe": "^17.6.0",
```
Payload uses version 10 :(

myniqx commented:
to try it locally
- download cli https://docs.stripe.com/stripe-cli
- login with cli `./stripe login`
- listen with cli `./stripe listen --forward-to=localhost:3001/api/webhook`
- cli ll give ur webhook secret, you must add .env file both stripes secrets.
- then run hvn and go donate component. 4242 4242 4242 4242 is valid test card. enter the rest what ever u wish.
- if donatíon is succesful you can see details in localhost/admin

shadcn ui has some serius style issues. i add tab component and it doest see its styles. 
other forms are not serius.

myniqx commented on apps/hvn/src/collections/Donations.ts:
```diff
@@ -0,0 +1,80 @@
+export const Donations: CollectionConfig = {
+  access: {
+    create: ({ req }) => {
+      return req.credentials === 'same-origin'
```
i dont know if it is work. admin cant add new donation. but webhook can add,
